protos/arm_authorizer_server/arm_authorizer_server.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package mavsdk.rpc.arm_authorizer;
+
+import "mavsdk_options.proto";
+
+option java_package = "io.mavsdk.arm_authorizer";
+option java_outer_classname = "ArmAuthorizerProto";
+
+service ArmAuthorizerServerService {
+    // Subscribe to arm authorization request messages. Each request received should respond to using RespondArmAuthorization
+    rpc SubscribeArmAuthorization(SubscribeArmAuthorizationRequest) returns(stream SubscribeArmAuthorizationResponse) { option (mavsdk.options.async_type) = ASYNC; }
+
+    // Authorize arm for the specific time
+    rpc AcceptArmAuthorization(AcceptArmAuthorizationRequest) returns(AcceptArmAuthorizationResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    // Reject arm authorization request
+    rpc RejectArmAuthorization(RejectArmAuthorizationRequest) returns(RejectArmAuthorizationResponse) { option (mavsdk.options.async_type) = SYNC; }
+}
+
+// Messages for SubscribeArmAuthorization
+message SubscribeArmAuthorizationRequest {}
+message SubscribeArmAuthorizationResponse {
+    uint32 system_id = 1; // vehicle system id
+}
+
+// Messages for RespondArmAuthorization
+message AcceptArmAuthorizationRequest {
+    int32 valid_time = 1; // Time in seconds for which this authorization is valid
+}
+
+// Result type
+message AcceptArmAuthorizationResponse {
+    CommandAnswer command_answer = 1; // Result enum value
+}
+
+// Messages for RespondArmAuthorization
+message RejectArmAuthorizationRequest {
+    bool temporarily = 1; // True if the answer should be TEMPORARILY_REJECTED, false for DENIED
+    RejectionReason reason = 2; // Reason for the arm to be rejected
+    int32 extra_info = 3; // Extra information specific to the rejection reason (see https://mavlink.io/en/services/arm_authorization.html)
```

Comment:
I don't find this extra info? Can you help me understand how that works?

Reply:
I'm not sure if `extra_info` is the best name for this field, actually. In the [arm authorization protocol description in MavLink](https://mavlink.io/en/services/arm_authorization.html), it is stated that `result_param2` field of the ACK message should be set this way:
>  result_param2=if result is ACCEPTED the it should be set with the time in seconds that this authorization is valid otherwise an aditional information about why it was denied should be set. example: for result_param1=MAV_ARM_AUTH_DENIED_REASON_INVALID_WAYPOINT or MAV_ARM_AUTH_DENIED_REASON_AIRSPACE_IN_USE it may have the index of the waypoint that caused it to be denied.

This is pretty ambiguous and not well defined, so I thought we could leave it to the user. So `result_param2` should be filled with the time interval in `AcceptArmAuthorization` function, but with this `extra_info` othersiwe. I can also see that in `px4` firmware, it is checked only if the reason is `MAV_ARM_AUTH_DENIED_REASON_INVALID_WAYPOINT`, exactly as descried in the example above

Reply:
Ah, got it.